lib/stanford-mods/origin_info.rb

```diff
@@ -38,16 +53,28 @@ def pub_date_best_single_facet_value(date_el_array)
         desired_el = self.class.keyDate(date_el_array)
         result = DateParsing.facet_string_from_date_str(desired_el.content) if desired_el
         return result if result
-        # settle for earliest parseable date -- should we care about encoding?
-        poss_results = {}
-        date_el_array.each { |el|
-          result = DateParsing.sortable_year_string_from_date_str(el.content)
-          poss_results[result] = el.content if result
-        }
-        earliest = poss_results.keys.sort.first if poss_results.present?
-        return DateParsing.facet_string_from_date_str(poss_results[earliest]) if earliest
+        # settle for earliest parseable date
+        earliest = self.class.earliest_date(date_el_array)
+        return DateParsing.facet_string_from_date_str(earliest[1]) if earliest.present?
+      end
+
+      # given the passed date elements, look for a single keyDate and use it if there is one;
+      #    otherwise pick earliest parseable date
+      # @param [Array<Nokogiri::XML::Element>] date_el_array the elements from which to select a pub date
+      # @return [String] single String containing publication year for lexical sorting
+      def pub_date_best_sort_str_value(date_el_array)
+        return if date_el_array.empty?
+        # prefer keyDate
+        desired_el = self.class.keyDate(date_el_array)
+        result = DateParsing.sortable_year_string_from_date_str(desired_el.content) if desired_el
+        return result if result
+        # settle for earliest parseable date
+        earliest = self.class.earliest_date(date_el_array)
+        return earliest[0] if earliest.present?
```

Comment:
i should note this is one of those stylistic things that different developers would each probably do differently and i'm not sure there is a "correct" way of doing it (possibly short of declaring a custom data structure via a new class and then setting and returning that, which is overkill for this case in my opinion)